alert-common/src/main/java/com/synopsys/integration/alert/common/action/ApiAction.java

```diff
@@ -27,29 +27,50 @@
 
 public abstract class ApiAction {
 
+    /**
+     * @throws AlertException Overriding classes may throw this exception
+     */
     public FieldModel beforeSaveAction(FieldModel fieldModel) throws AlertException {
```

Comment:
SonarCloud does not want a method to throw an exception if the body of code does not throw that exception. In our case we do this so that overriding methods can throw the exception. The Javadoc allows SonarCloud to ignore this.